specs/spec.md

```diff
@@ -0,0 +1,329 @@
+# @TODO
+- [ ] Link to parsers.
+- [ ] attestations
+- [ ] what else needs to be defined in the transport header?
+
+# Hobbits
+
+*Written by Dean Eigenmann <dean@status.im> & Rene Nayman <rene@whiteblock.io>* 
+
+## Table of Contents
+
+1. [Abstract](#abstract)
+2. [Flow](#flow)
+3. [Wire Protocol](#wire-protocol)
+    - [Messages](#messages)
+4. [Transport](#transport)
+5. [Protocols](#protocols)
+    - [RPC](#rpc)
+    - [Gossip](#gossip)
+    - [PING](#ping)
+6. [Implementations](#implementations)
+
+## Abstract
+
+Hobbits is a modular wire protocol which allows implementers to experiment with various application logic over a practical network in an agnostic and expedient manner.
+
+Hobbits wire protocol was developed so that Eth2.0 clients could begin network-level testing without requiring libp2p.
+
+## Flow
+
+We layout a network where the nodes are peered in the following manner:
+
+![nodes](./nodes.png)
+
+The following flow dictates how blocks are shared amongst peers:
+
+![sequence diagram](./diagram.png)
+
+1. Node 1 receives a new block `0xCAB` and gossips `hash(0xCAB)` to Node 2 and Node 3
```

Comment:
it might be worth it to define what this hashing algorithm is `hash(0xCAB)`

Reply:
@atoulme do we know what the hashing algo is yet? Otherwise I'll just pick one...

Reply:
We picked sha256 iirc (or maybe keccak). Pick something you like.